File: src/cpu.rs

// ===== Imports =====
use crate::{registers::Registers, clock::Clock};
// ===================

pub struct CPU {
  pub registers: Registers,
  pub clock: Clock,
}

impl CPU {
  pub fn new() -> Self {
    Self {
      registers: Registers::new(),
      clock: Clock::new(),
    }
  }
}

File: src/cpu.zig
const registers = @import("./cpu/registers.zig");
const memory = @import("./memory.zig");
const cartridge = @import("./cartridge.zig");

pub const CPU = struct {
    registers: registers.Registers,
    memory: memory.Memory,

    pub fn new(cart: cartridge.Cartridge) CPU {
        return CPU{
            .registers = registers.Registers.new(),
            .memory = memory.Memory.new(cart),
        };
    }
};
